src/overlays/actors/ovl_En_Ma1/z_en_ma1.c

```diff
@@ -19,45 +36,412 @@ const ActorInit En_Ma1_InitVars = {
     (ActorFunc)EnMa1_Update,
     (ActorFunc)EnMa1_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0690.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ma1/func_80AA0778.s")
+ColliderCylinderInit D_80AA1640 = {
+    0x0A, 0x00,       0x00, 0x39, 0x20,   0x01,   0x00,       0x00,   0x00,   0x00,   0x00,
+    0x00, 0x00000000, 0x00, 0x00, 0x00,   0x00,   0x00000000, 0x00,   0x00,   0x00,   0x00,
+    0x00, 0x00,       0x01, 0x00, 0x0012, 0x002E, 0x0000,     0x0000, 0x0000, 0x0000,
+};
+
+Sub98Init5 D_80AA166C = {
+    0x00, 0x0000, 0x0000, 0x0000, 0xFF,
+};
+
+struct_D_80AA1678 D_80AA1678[] = {
+    { 0x06000820, 1.0f, 0x00, 0.0f },
+    { 0x06000820, 1.0f, 0x00, -10.0f },
+    { 0x06008D64, 1.0f, 0x00, 0.0f },
+    { 0x06008D64, 1.0f, 0x00, -10.0f },
+};
+
+Vec3f D_80AA16B8 = { 800.0f, 0.0f, 0.0f };
+
+u32 D_80AA16C4[] = {
+    0x06001F18,
+    0x06002B18,
+    0x06002F18,
+};
+
+u32 D_80AA16D0[] = {
+    0x06001B18,
+    0x06002318,
+    0x06002718,
+};
+
+s32 D_80AA16DC = 0;
+
+extern AnimationHeader D_06000820;
+extern SkeletonHeader D_06008460;
+extern AnimationHeader D_06008D64;
+
+u16 EnMa1_GetText(GlobalContext* globalCtx, EnMa1* this) {
+    u16 faceReaction = Text_GetFaceReaction(globalCtx, 0x17);
+    if (faceReaction != 0) {
+        return faceReaction;
+    }
+    if (CHECK_QUEST_ITEM(QUEST_SONG_EPONA)) {
+        return 0x204A;
+    }
+    if (gSaveContext.eventChkInf[1] & 0x40) {
+        return 0x2049;
+    }
+    if (gSaveContext.eventChkInf[1] & 0x20) {
+        if ((gSaveContext.infTable[8] & 0x20)) {
+            return 0x2049;
+        } else {
+            return 0x2048;
+        }
+    }
+    if (gSaveContext.eventChkInf[1] & 0x10) {
+        return 0x2047;
+    }
+    if (gSaveContext.eventChkInf[1] & 4) {
+        return 0x2044;
+    }
+    if (gSaveContext.infTable[8] & 0x10) {
+        if (gSaveContext.infTable[8] & 0x800) {
+            return 0x2043;
+        } else {
+            return 0x2042;
+        }
+    }
+    return 0x2041;
+}
+
+s16 func_80AA0778(GlobalContext* globalCtx, EnMa1* this) {
+    s16 ret = 1;
+    switch (func_8010BDBC(&globalCtx->msgCtx)) {
+        case 2:
+            switch (this->actor.textId - 0x2041) {
```

Comment:
This looks like it could just be a `switch (this->actor.textId)`, with all cases being properly shifted (`case 0x2041` instead of `case 0`, etc)

Reply:
I just tried this and it doesn't seem to work.